packages-exp/auth-exp/src/core/user/additional_user_info.test.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { IdTokenResponse } from '../../model/id_token';
+import { ProviderId } from '../providers';
+import { fromIdTokenResponse } from './additional_user_info';
+
+describe('core/user/additional_user_info', () => {
+
+  describe('fromIdTokenResponse', () => {
+    const profile = {login: 'scott', friends: [], netWorth: 5.00};
+    const rawUserInfo = JSON.stringify(profile);
+
+    describe('parses federated IDP response tokens', () => {
+
+      it('for FacebookAdditionalUserInfo', () => {
+        const additionalUserInfo = fromIdTokenResponse(idTokenResponse({providerId: ProviderId.FACEBOOK, rawUserInfo}))!;
+        const {isNewUser, providerId, username, profile} = additionalUserInfo;
+        expect(isNewUser).to.be.false;
+        expect(providerId).to.eq(ProviderId.FACEBOOK);
+        expect(username).to.be.null;
+        expect(profile).to.eq(profile);
```

Comment:
should this one be to.eql since you're doing value comparison?

Reply:
haha oops... It also makes no sense to compare it with itself!